pkg/api/v1/resources/core/metadata_extensions.go

```diff
@@ -17,3 +17,5 @@ func (m *Metadata) Ref() *ResourceRef {
 func (m *Metadata) Match(ref *ResourceRef) bool {
 	return m.GetNamespace() == ref.GetNamespace() && m.GetName() == ref.GetName()
 }
+
+type Predicate func(metadata *Metadata) bool
```

Comment:
nit: Since this is in the `core` package, `core.Predicate` is kind of generic thing. Perhaps we clarify this is a `MetadataPredicate`?